src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/WorkspaceProjectContextProvider.cs

```diff
@@ -69,19 +67,17 @@ public async Task ReleaseProjectContextAsync(IWorkspaceProjectContextAccessor ac
 
         private async Task<IWorkspaceProjectContext?> CreateProjectContextHandlingFaultAsync(ProjectContextInitData data, object? hostObject)
         {
-            // TODO: https://github.com/dotnet/project-system/issues/353.
-            await _threadingService.SwitchToUIThread();
-
             try
             {
                 // Call into Roslyn to init language service for this project
-                IWorkspaceProjectContext context = _workspaceProjectContextFactory.Value.CreateProjectContext(
+                IWorkspaceProjectContext context = await _workspaceProjectContextFactory.Value.CreateProjectContextAsync(
                                                                                     data.LanguageName,
                                                                                     data.WorkspaceProjectContextId,
                                                                                     data.ProjectFilePath,
                                                                                     data.ProjectGuid,
                                                                                     hostObject,
-                                                                                    data.BinOutputPath);
+                                                                                    data.BinOutputPath,
+                                                                                    CancellationToken.None);
```

Comment:
I don't believe we have anything to pass here for cancellation.

Reply:
that's fine. we'll continue to have the parameter though in case this is something you hookup in the future.  for example, i could imagine a world where the user decides to close VS while we're still loading everything.  This would allow youto thread though cancellation if you wanted us to abort.

Our positionhere is: we will support cancellation up through the point where we're still doing computation but haven't done any mutation.  once we mutate our actual internal state, we ignore caller-requested cancellation and finish what we're doing.  That way our internal models are always in a consistent state.